tendermint/src/lite.rs

```diff
@@ -1,7 +1,12 @@
 //! Core logic and traits of a light client.
 
+// TODO: don't expose the verifier, just the types and the public API.
+// the verifier has low level functions that can be used incorrectly.
+
+pub mod public;
```

Comment:
I think for now it is cool to have that distinction and naming (public vs the rest). We should come up with another name though (maybe `verification`?) and not export the other modules which contain methods that can be used incorrectly.

Reply:
Yep, just didn't get around to unexposing the other pieces. Are you suggesting we have a `verifier` and a `verification` module and only expose the latter ? I also don't think it'd be that bad if we consolidated the verifier.rs and public.rs into a single `verifier` module and just expose only the specific functions in `lite.rs`

Reply:
> Are you suggesting we have a verifier and a verification module and only expose the latter ? 

Nope, I just think `public` is confusing and there should only be one module dealing with verification (`verifier` or `verification`).

>  I also don't think it'd be that bad if we consolidated the verifier.rs and public.rs into a single verifier module and just expose only the specific functions in lite.rs

Yes, that sounds right! 👍 (and was more what I meant, too)